src/main/java/com/google/songgame/servlets/UserServlet.java

```diff
@@ -49,9 +53,61 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     datastore.put(userEntity);
   }
 
+  @Override
+  public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Map<String, String> roomProperties = readJSONFromRequest(request);
+
+    Query userQuery =
+        new Query("User").addFilter("userId", FilterOperator.EQUAL, roomProperties.get("userId"));
+    PreparedQuery result = datastore.prepare(userQuery);
+    Entity currentUser = result.asSingleEntity();
+
+    // Add current roomId to current user.
+    currentUser.setProperty("roomId", roomProperties.get("roomId"));
+
+    datastore.put(currentUser);
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Cookie[] cookies = request.getCookies();
+
+    // Check if current cookie value exists
+    if (cookies != null) {
+      String userId = getUserId(request);
+      for (Cookie userCookie : cookies) {
+        if (userCookie.getValue().equals(userId)) {
+          response.sendRedirect("/join-room.html");
+          return;
+        }
+      }
```

Comment:
Curious as to why this loop is here. If the userId exists, why can't you immediately redirect?